longintrepr.h missing on Python 3.11

I'm upgrading a project to Python 3.11, mostly for the amazing error messages it provides. I now get an error from pip,

`      ushuffle_cython.c:215:12: fatal error: longintrepr.h: No such file or directory
        215 |   #include "longintrepr.h"
            |            ^~~~~~~~~~~~~~~
      compilation terminated
`

This seems to have affected a few other projects around the net, and it's definitely caused by upgrading to Python 3.11. I tried pointing to the new location of longintrepr.h, but got other build errors, so I'm assuming there will have to be several fixes. Let me know if there's more info I can provide.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,0 +1,3 @@
+[build-system]
+requires = ["setuptools", "wheel", "cython>=0.29"]
+build-backend = "setuptools.build_meta"

